cloud/pkg/devicecontroller/devicecontroller.go

```diff
@@ -58,6 +61,10 @@ func (dc *DeviceController) Enable() bool {
 
 // Start controller
 func (dc *DeviceController) Start() {
+	if !dc.Enable() {
+		klog.Error("DeviceController is disabled, failed to start")
+		return
+	}
```

Comment:
No need to return here, for the disabled module, which will not be started.

Reply:
The log also can be deleted, when disabled, the Start will not be called

Reply:
@fisherxu, delete these three lines or just return with no log? 
and I think it is the last insurance if someone start disabled module, or process will panic "nil pointer" error at later lines.

Reply:
Only start the enabled module. https://github.com/kubeedge/kubeedge/blob/master/staging/src/github.com/kubeedge/beehive/pkg/core/core.go#L18

Reply:
@fisherxu well, get it

Reply:
@fisherxu done